### DISCUSSION TITLE
SAML IdP certificate scoped for the application integration

### DISCUSSION THREAD

**Discussion by mariojuridiaz** [keycloak/keycloak#14599] [Q&A]
Hello,

We are integrating a RP via SAML using IDP-Initiated SSO and we would like to know if Keycloak has a feature which allow us to generate an alternative IdP certificate and signing keys scoped only for the application integration (client in Keycloak), instead of using realm certificate and signing keys. 

For example, other IdP's like Okta or Jumpcloud, allows you to choose between a certificate and signing keys scoped to application integration or scoped to organization.

This is due to our RP has several performance issues and we can not use the IdP realm certificate.

Any idea or suggestion is welcome.

Thanks.

**Comment by Szer** [comment]
@mariojuridiaz were you able to workaround that? I have the same issue at the moment where one of our customers requires CA-trusted certificate even for assertion signing, but I can't change active certificate in KC because it will break dozens of other SAML SSO connections
